src/main/java/training_day5/HW/Student.java

```diff
@@ -0,0 +1,95 @@
+package training_day5.HW;
+
+import javax.security.auth.Subject;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+public class Student extends Person{
+
+    private int clas;
+    private int numberInClass;
+    private School school;
+    private HashMap<Subjects, Grade> gradesForSubject = new HashMap<>();
+    private List<Subjects> subjects = new ArrayList<>();
+
+    public Student(String name, List<Subjects> subjects, int clas, int numberInClass, School school){
+        super(name);
+        school.registerStudent(this);
+        this.subjects.addAll(subjects);
+        this.clas = clas;
+        this.numberInClass = numberInClass;
+        this.school = school;
+    }
+
+    public int getClas() {
+        return clas;
+    }
+
+    public void setClas(int clas) {
+        this.clas = clas;
+    }
+    public int getNumberInClass(){
+        return this.numberInClass;
+    }
+    public void setNumberInClass(int number){
+        this.numberInClass = number;
+    }
+
+    public School getSchool(){
+        return this.school;
+    }
+
+    public void setSchool(School school){
+        this.school = school;
+    }
+
+    public boolean addNewGrade(Subjects subject, Grade grade){
+        if(grade.teacher.getSubjects().contains(subject) && this.subjects.contains(subject) &&
+            grade.teacher.getSchool() == this.school && grade.value >= 2 && grade.value <= 6){
+            this.gradesForSubject.put(subject, grade);
+            return true;
+        }
+        else
+            return false;
+    }
+
+    public void updateGrade(Subjects subject, int value){
+
+        Grade temp = this.gradesForSubject.get(subject);
```

Comment:
You don't have to use this.variableName if you don't have a method parameter with the same name as the one in the class.